static/dependencies.config.php

```diff
@@ -19,266 +20,237 @@
 		'constructParams' => [
-			[Dice::INSTANCE => '$basepath'],
+			$basepath,
 		],
 	],
-	\Psr\Log\LoggerInterface::class                                    => [
+	\Psr\Log\LoggerInterface::class => [
 		'instanceOf' => \Friendica\Core\Logger\Factory\Logger::class,
-		'call'       => [
+		'call' => [
 			['create', [], Dice::CHAIN_CALL],
 		],
 	],
-	\Friendica\Core\Logger\Type\SyslogLogger::class                    => [
+	\Friendica\Core\Logger\Type\SyslogLogger::class => [
 		'instanceOf' => \Friendica\Core\Logger\Factory\SyslogLogger::class,
-		'call'       => [
+		'call' => [
 			['create', [], Dice::CHAIN_CALL],
 		],
 	],
-	\Friendica\Core\Logger\Type\StreamLogger::class                    => [
+	\Friendica\Core\Logger\Type\StreamLogger::class => [
 		'instanceOf' => \Friendica\Core\Logger\Factory\StreamLogger::class,
-		'call'       => [
+		'call' => [
 			['create', [], Dice::CHAIN_CALL],
 		],
 	],
 	\Friendica\Core\Logger\Capability\IHaveCallIntrospections::class => [
-		'instanceOf'      => \Friendica\Core\Logger\Util\Introspection::class,
+		'instanceOf' => \Friendica\Core\Logger\Util\Introspection::class,
 		'constructParams' => [
 			\Friendica\Core\Logger\Capability\IHaveCallIntrospections::IGNORE_CLASS_LIST,
 		],
 	],
-	'$devLogger'                                                       => [
+	'$devLogger' => [
 		'instanceOf' => \Friendica\Core\Logger\Factory\StreamLogger::class,
-		'call'       => [
+		'call' => [
 			['createDev', [], Dice::CHAIN_CALL],
 		],
 	],
-	Cache\Capability\ICanCache::class => [
-		'instanceOf' => Cache\Factory\Cache::class,
-		'call'       => [
+	\Friendica\Core\Cache\Capability\ICanCache::class => [
+		'instanceOf' => \Friendica\Core\Cache\Factory\Cache::class,
+		'call' => [
 			['createLocal', [], Dice::CHAIN_CALL],
 		],
 	],
-	Cache\Capability\ICanCacheInMemory::class => [
-		'instanceOf' => Cache\Factory\Cache::class,
-		'call'       => [
+	\Friendica\Core\Cache\Capability\ICanCacheInMemory::class => [
+		'instanceOf' => \Friendica\Core\Cache\Factory\Cache::class,
+		'call' => [
 			['createLocal', [], Dice::CHAIN_CALL],
 		],
 	],
-	Lock\Capability\ICanLock::class => [
-		'instanceOf' => Lock\Factory\Lock::class,
-		'call'       => [
+	\Friendica\Core\Lock\Capability\ICanLock::class => [
+		'instanceOf' => \Friendica\Core\Lock\Factory\Lock::class,
+		'call' => [
 			['create', [], Dice::CHAIN_CALL],
 		],
 	],
-	App\Arguments::class => [
-		'instanceOf' => App\Arguments::class,
+	\Friendica\App\Arguments::class => [
+		'instanceOf' => \Friendica\App\Arguments::class,
 		'call' => [
-			['determine', [$_SERVER, $_GET], Dice::CHAIN_CALL],
+			['determine', [$serverVars, $getVars], Dice::CHAIN_CALL],
 		],
 	],
 	\Friendica\Core\System::class => [
 		'constructParams' => [
-			[Dice::INSTANCE => '$basepath'],
+			$basepath,
 		],
 	],
-	App\Router::class => [
+	\Friendica\App\Router::class => [
 		'constructParams' => [
-			$_SERVER,
+			$serverVars,
 			__DIR__ . '/routes.config.php',
```

Comment:
Shouldn't we use the `$basepath` variable here instead for better portability?

Reply:
At the moment `__DIR__` tells us, that the `router.config.php` is in the same folder.

Using `$basepath` here would require us to add the name of the current `static` folder:

```php
$basepath . '/static/routes.config.php',
```

This would add a "dependency" to the folder name that I would like to avoid for better portability.

Reply:
But this, buried pretty deep into the file, requires that the `routes.config.php` be in the same folder as `dependencies.config.php`. With `$basepath` instead, we'd be able to search for the relative path of `routes.config.php` and get a hit.

I've read in this thread that you'd like to move `dependencies.config.php` into a (the?) config folder, will you remember this dependency between both files?

Reply:
I have no strong opinion on this topic.

My point is that if we will rename the `static` folder into `config` (or something similar) we will not have to change the dependencies inside this folder. Another point is that this line is out of scope of this PR, because it was defined this way before and there is nothing wrong with it. 

If you like I will change the path to use `$basepath`, but I simple don't see a need for this at the moment.

Reply:
I see your point. There already is a `/config` folder, however it's been meant for server-specific config, not for the static configuration like the dependencies and the routes.

Please don't change with the base path.